cloudinit/sources/DataSourceLXD.py

```diff
@@ -331,9 +348,16 @@ def read_metadata(
                 md[LXD_SOCKET_API_VERSION]["config"][cfg_key] = response.text
                 # Promote common CONFIG_KEY_ALIASES to top-level keys.
                 if cfg_key in CONFIG_KEY_ALIASES:
-                    md[CONFIG_KEY_ALIASES[cfg_key]] = response.text
+                    # Due to sort of config_routes, promote cloud-init.*
+                    # aliases before user.*. This allows user.* keys to act as
+                    # fallback config on old LXD, with new cloud-init images.
+                    if CONFIG_KEY_ALIASES[cfg_key] not in md:
```

Comment:
If I pass `cloud-init.user-data` and `user.user-data` simultaneously, then we'll ignore `user.user-data` in favor of `cloud-init.user-data`? If so, I think we should at least raise a warning if both are passed simultaneously.